pennylane/__init__.py

```diff
@@ -338,3 +307,45 @@ def run_cnot():
 def version():
     """Returns the PennyLane version number."""
     return __version__
+
+
+# add everything as long as it's not a module and not prefixed with _
+_all = sorted(
+    [
+        name
+        for name, function in globals().items()
+        if not (name.startswith("_") or isinstance(function, types.ModuleType))
+    ]
+)
+
+
+_qchem = None
+
+
+def __getattr__(name):
+    """Ensure that the qchem module is imported lazily"""
+    if name == "qchem":
+        global _qchem  # pylint: disable=global-statement
```

Comment:
For my understanding: how come we have to define `_qchem` as global here and not on line 322?

Reply:
Ah, `_qchem` is an existing module level variable, but the way Python scope works is as follows:

```python
a = 5

def f():
    # No assignment; Python assumes
    # a refers to the global variable a above.
    print(a)

def g():
    # a is assigned to within the function;
    # Python assigns it with a scope *local*
    # to the function, independent of the global
    # a above.
    a = 7
    print(a)
```

If we wish `g()` to assign value to the _global_ variable `a` above, we need to then explicitly state this using the `global` keyword.